common/models/application.js

```diff
@@ -81,8 +81,14 @@ module.exports = function(Application) {
    * A hook to generate keys before creation
    * @param next
    */
-  Application.beforeCreate = function(next) {
-    var app = this;
+  Application.observe('before save', function(ctx, next) {
+    if (!ctx.instance) {
+      // Example: Application.updateAll() and Application.updateOrCreate()
+      var msg = 'Application object does not support data-based updates';
+      return next(new Error(msg));
```

Comment:
There is already a unit-test for this and it's passing: https://github.com/strongloop/loopback/blob/baf372b8d7f1bd578759e11e596ec7d633b70cb5/test/model.application.test.js#L97-L120